src/Namshi/Innovate/Client.php

```diff
@@ -167,6 +181,35 @@ public function createMpiRequest($method = 'GET', $uri = null, $headers = null,
     }
 
     /**
+     * Given a cart id reference will search for transaction for that cart id on innovate
+     *
+     * @param $ref
+     * @return \SimpleXMLElement
+     * @throws InnovateException
+     */
+    public function searchPaymentsByCartId($ref)
+    {
+        $client = new \Guzzle\Http\Client(self::INNOVATE_SEARCH_URL, array(
+            'request.options' => array(
+                'auth'    => array($this->merchantId, $this->key, 'Basic'),
+            )
+        ));
+        $request = $client->get('/tools/api/xml/transaction/'.$ref.'/cart');
```

Comment:
const URI_TRANSACTION_BY_CARTID = "'/tools/api/xml/transaction/%s/cart'"
and then
$client->get(sprintf(self::URI_TRANSACTION_BY_CARTID, $ref);